client/src/AddVideo.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+
+function AddVideo({ videos, setVideos }) {
+  const [showValid, Setvalid] = useState(true);
+    const [newVideo, setNewVideo] = useState({});
+   const [videoTitle, setTitle] = useState("");
+  const [videoUrl, setUrl] = useState("");
+  //   const [, setVideos] = useState(videos);
+
+  const handleShowValid = () => {
+    Setvalid(!showValid);}
+  
+   function isValidUrl(url) {
+    try {
+      new URL(url);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+  const handleAddVideo = (event) => {
+    event.preventDefault();
+    videoTitle !== ""
+      ? (newVideo.title = videoTitle)
+      : alert("Please add a title for your video.");
+
+    isValidUrl(videoUrl)
+      ? (newVideo.url = videoUrl)
+      : alert("Please add a valid url.");
+    newVideo.rating = Math.floor(Math.random() * 5000) + 1;
+
+ 
+ newVideo.id = videoUrl.split("=")[1];
+    setVideos([newVideo, ...videos]);
+     setNewVideo({});
+    setTitle("");
+    setUrl("");
+  };
+
+  return (
+    <div>
+      
+        <form onSubmit={handleAddVideo}>
+          <div>
+            <label htmlFor="title">Tile:</label>
+            <input
+              type="text"
+              id="title"
+              onChange={(e) => {
+                setTitle(e.target.value);
+              }}
+            ></input>
+          </div>
```

Comment:
This `div, label, input` pattern is repeated a couple times here; you coudl make this a React component and reuse it here instead